[Fixed] Added Strings #587

ayaan-qadri opened the pull request:
<!--
Hello! Thank you for contributing to Code Thesaurus!

This template can help you fill out a pull request with all the information 
needed to review it. The text between these exclamation tags are comments 
and won't show up on the pull request. The text after ## are headers. Please
leave the headers unless otherwise noted and follow the instructions to 
fill out each section with as much information as you can to assist the 
reviewer!
-->


## What GitHub issue does this PR apply to?

 <!--
   If you're working on an existing issue, replace "xxxx" below with the issue 
   number. You can change it to say only "Closes #", "Fixes #", or "Resolves #".
   Don't add the word "issue" to it otherwise it won't link correctly.

   If you got here via the edit buttons on the website and/or your changes 
   don't relate to an issue, feel free to delete "Resolves #" and put "None", 
   or note why you chose to make a PR!
   -->

Resolves #587 


## What changed and why?

   <!-- Please replace this line with a description of your changes. -->


## (If editing Django app) Please add screenshots

   <!--
   If this doesn't apply, you can delete this header and section.
   If it's applicable, please copy screenshots, then paste them below.
   GitHub will paste images as a line of text as: ![image](url)
   You can preview it with the "Preview" button above.
   -->


## Checklist

   <!-- 
   Each - [ ] below is a checkbox that will show up on the PR.
   Either add an X inside the [X], or submit the PR and click the checkboxes.
   -->

- [x] I claimed any associated issue(s) and they are not someone else's
- [x] I have looked at documentation to ensure I made any revisions correctly
- [x] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes


## Any additional comments or things to be aware of while reviewing?

   <!-- Please replace this line with any extra information that's helpful -->

ayaan-qadri committed:
Added Strings #587

ayaan-qadri commented:
I have tested it also by running it locally, Works fine for me in well formated!!

cafce25 commented on web/thesauruses/lua/5/strings.json:
```diff
@@ -0,0 +1,272 @@
+      },
+      "max_length_of_string": {
+        "name": "Maximum length of string",
+        "code": "230,000 characters"
```
Not sure where you got this number from, I don't believe there is a limit so low, and I can create larger strings on my machine, I think it's either capped by ram or the size of an integer.

geekygirlsarah replied:
I think @cafce25 is right. See https://lua-users.org/lists/lua-l/2007-10/msg00130.html

cafce25 commented on web/thesauruses/lua/5/strings.json:
```diff
@@ -0,0 +1,272 @@
+      },
+      "assign_new_string": {
+        "name": "Assign string from another string",
+        "code": "a = 'a'\na = 'b'"
```
I think this concept is about copying string values, not creating multiple variables from different string literals.

cafce25 commented on web/thesauruses/lua/5/strings.json:
```diff
@@ -0,0 +1,272 @@
+      },
+      "length_of_string": {
+        "name": "Length of string",
+        "code": "string.len('string')"
```
Probably add the method syntax
```suggestion
        "code": [
            "'string':len()",
            "string.len('string')"
        ]

```

geekygirlsarah replied:
Perhaps to make it less confusing, do something like `'my text':len()` and `string.len('my text')` to help distinguish string (a variable) from string (text) vs string (a class)

cafce25 commented on web/thesauruses/lua/5/strings.json:
```diff
@@ -0,0 +1,272 @@
+      },
+      "is_empty": {
+        "name": "Is string empty?",
+        "code": "str = ''\nif (str == nil or str == '') then\nprint('str is empty')\nend"
```
Can you inden't the code inside the `if`, also there is a more readable way to put multiple lines like so:
```suggestion
        "code": [
            "str = ''",
            "if (str == nil or str == '') then",
            "    print('str is empty')",
            "end"
        ]
```

ayaan-qadri replied:
I keep it without  indent because in lau we do not need to specify indent.

cafce25 commented on web/thesauruses/lua/5/strings.json:
```diff
@@ -0,0 +1,272 @@
+      },
+      "concatenate_many_strings": {
+        "name": "Concatenate/join many strings",
+        "code": "str = 'Hello'\nvar2 = ' World '\nvar3 = str..var2\nprint(str..var2..var3)",
```
I'd love to see `table.concat` here too.

ayaan-qadri replied:
Added

cafce25 commented on web/thesauruses/lua/5/strings.json:
```diff
@@ -0,0 +1,272 @@
+      },
+      "format_number_with_positive_negative_sign": {
+        "name": "Format number with positive/negative signs",
+        "not-implemented": true
```
What about `"%+d"` ?

ayaan-qadri replied:
Done!

cafce25 commented on web/thesauruses/lua/5/strings.json:
```diff
@@ -0,0 +1,272 @@
+        "name": "Format number into hexadecimal",
+        "code": "str = '10'\nprint(string.format('%x',str))",
+        "comment":"%x for lowercase letters.\n%X upper letters."
+      }
```
You're using string literals for all these format number as string concepts. I think the point is to show how to convert numbers to strings and this might be confusing.

ayaan-qadri replied:
Same syntax will also work for number !

cafce25 replied:
You know that and I do but someone new to the language might not!
Since these concepts are about converting numbers to strings and not strings to strings the examples should use numbers imo

ayaan-qadri replied:
Alright, I'll add comment and will change it.

cafce25 reviewed (changes requested):
Hey,
thanks for taking the time to write this up. This already looks quite good, except for the few things I pointed out.
I'd also think many of the concepts can be more succinctly written using method syntax like I pointed out on `length_of_string`.
Other than that I think you're well on your way to becoming a contributor! Yay!

cafce25 commented:
If you edit this:
> Resolves #xxxx

to 

> Resolves #587

Github will work magic and link the issue to this PR and close the issue upon merging.

geekygirlsarah committed:
Merge branch 'main' into issue-587-fix-Lua_add_strings

geekygirlsarah reviewed (changes requested):
I don't know enough Lua to really know the correct ways to do things, but @cafce25 points out some good points. I think if you update those, this should be probably good to merge!

ayaan-qadri committed:
Apply suggestions from code review

Co-authored-by: Jonathan Birk <1965620+cafce25@users.noreply.github.com>

ayaan-qadri committed:
Update according suggested changes

ayaan-qadri commented:
Changed and tested, Works fine !!

ayaan-qadri committed:
Merge branch 'main' into issue-587-fix-Lua_add_strings

ayaan-qadri committed:
Update strings.json

some values of variable was string that changed into integer.

ayaan-qadri committed:
Update strings.json

Added "\ntab = {'codethesaur.us',' is',' great',' learning tool!'}\nprint(table.concat(tab))" at wrong section, corrected.

ayaan-qadri commented:
i guess, It is now ready to merge !

ayaan-qadri committed:
Update strings.json

solved this: ```Error:  `lua/5/strings.json`, ID: `import` is confusing: `code` is empty but there's no `not-implemented` either```

geekygirlsarah committed:
Merge branch 'main' into issue-587-fix-Lua_add_strings

ayaan-qadri reviewed (commented):
All changes have done !

ayaan-qadri reviewed (commented):
Merged into main.

cafce25 commented:
Hey @PentW0lf,

I've seen you changed things but was not able to review anything yet.
Since everyone working on this is doing it on their free time sometimes things can take a 
while and you'll have to wait for them to find some time to review/merge.

By the way, you constantly pinging this PR rather than getting me to work on it faster has rather
the opposite effect by slowing me down due to constant new messages in my inbox.

Please be patient and wait, I'm sure Sarah will do her best to merge this as fast as possible.

ayaan-qadri commented:
> 

my fault, idk when and how did i constantly ping you, I'm beginner tho, can you please tell how did you get ping by me so that i keep it in mind ?, btw i didn't mention you anywhere nor i comment

cafce25 commented:
Every time you write a message / change the requested reviewer. Everybody watching this repo / PR gets a message in their inbox.

ayaan-qadri commented:
> 

Pls unsubscribe so that you do not get mail/notification because changes and some messages/comment on changes are required to contribute the project so you don not get distribution on your work.

geekygirlsarah commented:
> Pls unsubscribe so that you do not get mail/notification because changes and some messages/comment on changes are required to contribute the project so you don not get distribution on your work.

It's my project, so it really only makes sense for me to stay subscribed since that's how I learn there's updates. But like @cafce25 said, it's a side project for me, and between a full time job and also mentoring a high school robotics team, I definitely have a lot going on. But I do get to the PRs and issues, it's just not always immediate.

geekygirlsarah committed:
Merge branch 'main' into issue-587-fix-Lua_add_strings

geekygirlsarah reviewed (approved):
Your updates look good! I can merge it in.

Congrats on your first Code Thesaurus contribution! 🎉 

Have a great weekend!

ayaan-qadri commented:
> > Pls unsubscribe so that you do not get mail/notification because changes and some messages/comment on changes are required to contribute the project so you don not get distribution on your work.
> 
> It's my project, so it really only makes sense for me to stay subscribed since that's how I learn there's updates. But like @cafce25 said, it's a side project for me, and between a full time job and also mentoring a high school robotics team, I definitely have a lot going on. But I do get to the PRs and issues, it's just not always immediate.

yup, It's your project, i never said to merge PR immediate. That guy said that i was pinging him continuously that i didn't so asked  him how was you getting constantly  ping by me

he said "Every time you write a message / change the requested reviewer. Everybody watching this repo / PR gets a message in their inbox." (as you can see above).

so does it mean should i stop making changes on your project that are required ?,do i stop contribute ?

AS solution/Suggestion i said him to unsubscribe(for this particular PR/issue) so that he don't get notification further if i do changed or put message/comment on required changes.

<img src="https://i.imgur.com/pR2uLbk.png">

Again i never told to merge instantly nor i have intention to ping him constantly.
